src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -384,6 +387,10 @@ private function getDefinitionSchema(string $resourceClass, ResourceMetadata $re
             $normalizedPropertyName = $this->nameConverter ? $this->nameConverter->normalize($propertyName) : $propertyName;
 
             if ($propertyMetadata->isRequired()) {
+                if (false === $propertyMetadata->isWritable()) {
+                    throw new RuntimeException(sprintf('The property "%s" of the resource "%s" can not be required and read-only at the same time.', $propertyName, $resourceClass));
```

Comment:
Our metadata system does not provide enough flexibility to do this correctly. Let's say if writable is set to `false` after `ValidatorPropertyMetadataFactory`, then what?